crates/handlers/src/admin/v1/users/set_password.rs

```diff
@@ -0,0 +1,190 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use aide::{transform::TransformOperation, NoApi, OperationIo};
+use axum::{extract::State, response::IntoResponse, Json};
+use hyper::StatusCode;
+use mas_storage::BoxRng;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use ulid::Ulid;
+use zeroize::Zeroizing;
+
+use crate::{
+    admin::{call_context::CallContext, params::UlidPathParam, response::ErrorResponse},
+    impl_from_error_for_route,
+    passwords::PasswordManager,
+};
+
+#[derive(Debug, thiserror::Error, OperationIo)]
+#[aide(output_with = "Json<ErrorResponse>")]
+pub enum RouteError {
+    #[error(transparent)]
+    Internal(Box<dyn std::error::Error + Send + Sync + 'static>),
+
+    #[error("Password hashing failed")]
+    Password(#[source] anyhow::Error),
+
+    #[error("User ID {0} not found")]
+    NotFound(Ulid),
+}
+
+impl_from_error_for_route!(mas_storage::RepositoryError);
+
+impl IntoResponse for RouteError {
+    fn into_response(self) -> axum::response::Response {
+        let error = ErrorResponse::from_error(&self);
+        let status = match self {
+            Self::Internal(_) | Self::Password(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::NotFound(_) => StatusCode::NOT_FOUND,
+        };
+        (status, Json(error)).into_response()
+    }
+}
+
+fn password_example() -> String {
+    "hunter2".to_owned()
+}
+
+/// # JSON payload for the `POST /api/admin/v1/users/:id/set-password` endpoint
+#[derive(Deserialize, JsonSchema)]
+#[schemars(rename = "SetUserPasswordRequest")]
+pub struct Request {
+    /// The password to set for the user
+    #[schemars(example = "password_example")]
+    password: String,
+}
+
+pub fn doc(operation: TransformOperation) -> TransformOperation {
+    operation
+        .id("setUserPassword")
+        .summary("Set the password for a user")
+        .tag("user")
+        .response_with::<200, StatusCode, _>(|t| t.description("Password was set"))
+        .response_with::<404, RouteError, _>(|t| {
+            let response = ErrorResponse::from_error(&RouteError::NotFound(Ulid::nil()));
+            t.description("User was not found").example(response)
+        })
+}
+
+#[tracing::instrument(name = "handler.admin.v1.users.set_password", skip_all, err)]
+pub async fn handler(
+    CallContext {
+        mut repo, clock, ..
+    }: CallContext,
+    NoApi(mut rng): NoApi<BoxRng>,
+    State(password_manager): State<PasswordManager>,
+    id: UlidPathParam,
+    Json(params): Json<Request>,
+) -> Result<StatusCode, RouteError> {
+    let user = repo
+        .user()
+        .lookup(*id)
+        .await?
+        .ok_or(RouteError::NotFound(*id))?;
+
```

Comment:
I was assuming that because this is the admin API, it should just accept it? Synapse [does that](https://github.com/element-hq/synapse/blob/4580e12275793da77edca7dccdf778e53ca31f4a/synapse/rest/admin/users.py#L412) (and I don't think it checks for empty passwords either)

But I'd be happy to make it check here and have a `skip_password_check` parameter or something similar